packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -144,10 +145,19 @@ export default createRule<[], MessageId>({
                 ]),
               ];
 
-              const stringValue = getStaticStringValue(expression);
-
-              if (stringValue != null) {
-                const escapedValue = stringValue.replace(/([`$\\])/g, '\\$1');
+              if (isLiteral(expression)) {
+                const escapedValue =
+                  typeof expression.value === 'string'
+                    ? // '1'   -> 1
+                      // '`'   -> \`
+                      // '${}' -> \${}
+                      // '\\'  -> \\
+                      expression.raw.slice(1, -1).replace(/([`$])/g, '\\$1')
```

Comment:
Should we only escape `${` instead of all `$`? Also what if these characters are already escaped?

Reply:
Oops, I see @kirkwaiblinger has come up with the exact same ideas! My suggestion for the "already-escaped" case is to do something like `(?<=(\\\\)*)`.